tests/test_lowlevel.py

```diff
@@ -204,3 +203,26 @@ def test_use_proxy_from_environment(httpbin, var, scheme):
 
         # it had actual content (not checking for SOCKS protocol for now)
         assert len(fake_proxy.handler_results[0]) > 0
+
+
+def test_redirect_rfc1808_to_non_ascii_location():
+    def redirect_resp_handler(sock):
+        consume_socket_content(sock, timeout=0.5)
+        sock.send(
+            b'HTTP/1.1 301 Moved Permanently\r\n'
+            b'Location: //%s:%d/?\xc5\xa0\r\n'
```

Comment:
Unfortunately, neither `%` or `format` work on byte strings in Python 3 prior to 3.5. This means this string will need to be constructed another way. Either using `join` on multiple byte strings, or the `+` operator will probably be the best way to approach this for all Python versions.